diff --git a/plex/CMakeModules/FindFriBiDi.cmake b/plex/CMakeModules/FindFriBiDi.cmake
index 63f8694..b1e363b 100644
--- a/plex/CMakeModules/FindFriBiDi.cmake
+++ b/plex/CMakeModules/FindFriBiDi.cmake
@@ -22,6 +22,19 @@ FIND_PATH(xFRIBIDI_INCLUDE_DIR fribidi.h
   )
 set(FRIBIDI_INCLUDE_DIR ${xFRIBIDI_INCLUDE_DIR})
 
+FIND_PATH(xGLIB_INCLUDE_DIR glib.h
+  PATHS /usr/local/include
+        /usr/include
+  PATH_SUFFIXES glib-2.0
+  )
+SET(FRIBIDI_INCLUDE_DIR "${FRIBIDI_INCLUDE_DIR};${xGLIB_INCLUDE_DIR}")
+SET(xGLIB_CONFIG_INCLUDE_DIR)
+FIND_PATH(xGLIB_CONFIG_INCLUDE_DIR glibconfig.h
+  PATHS /usr/lib
+  PATH_SUFFIXES glib-2.0/include
+  )
+SET(FRIBIDI_INCLUDE_DIR "${FRIBIDI_INCLUDE_DIR};${xGLIB_CONFIG_INCLUDE_DIR}")
+
 SET(FRIBIDI_NAMES ${FRIBIDI_NAMES} fribidi libfribidi)
 FIND_LIBRARY(FRIBIDI_LIBRARY
   NAMES ${FRIBIDI_NAMES}
@@ -29,9 +42,9 @@ FIND_LIBRARY(FRIBIDI_LIBRARY
   )
 
 IF (FRIBIDI_LIBRARY AND FRIBIDI_INCLUDE_DIR)
-  SET(CMAKE_REQUIRED_INCLUDES ${FRIBIDI_INCLUDE_DIR})
+  SET(CMAKE_REQUIRED_INCLUDES ${FRIBIDI_INCLUDE_DIR})
   SET(CMAKE_REQUIRED_LIBRARIES ${FRIBIDI_LIBRARY})
-  CHECK_SYMBOL_EXISTS(fribidi_utf8_to_unicode fribidi.h FOUND_fribidi_utf8_to_unicode)
+  CHECK_SYMBOL_EXISTS(fribidi_utf8_to_unicode "fribidi.h" FOUND_fribidi_utf8_to_unicode)
   CHECK_SYMBOL_EXISTS(fribidi_charset_to_unicode fribidi.h FOUND_fribidi_charset_to_unicode)
   if(FOUND_fribidi_charset_to_unicode)
     SET(FRIBIDI_LIBRARIES ${FRIBIDI_LIBRARY})
